03-rnaseq/ortholog-mapping_rnaseq_01_ensembl.Rmd

```diff
@@ -0,0 +1,466 @@
+human_mouse_mapped_df %>%
+  dplyr::group_by(human_ensembl) %>%
+  dplyr::count() %>%
+  # Sort by highest `n` which would be the human Ensembl ID with the most mapped
+  # mouse Ensembl IDs
+  dplyr::arrange(desc(n))
+```
+
+Looks like we have mapped IDs!
+
+Now, let's get an idea of how many mouse Ensembl IDs we have that were not mapped to human Ensembl IDs.
+
+```{r}
+sum(is.na(human_mouse_mapped_df$human_ensembl))
+```
+
+We have 18,801 NAs, which means we have 18,801 mouse Ensembl IDs out of the 64,816 in `human_mouse_mapped_df`, that were not mapped to human Ensembl IDs.
+This is okay because we do not expect everything to map across species.
+
+## Take a look at some multi-mappings
+
+If a mouse Ensembl gene ID maps to multiple human Ensembl IDs, the associated values will get duplicated.
+Let's look at the `ENSMUSG00000000290` example below.
+
+```{r}
+human_mouse_mapped_df %>%
+  dplyr::filter(Gene == "ENSMUSG00000000290")
+```
+
+On the other hand, if you were to look at the original data associated to the mouse Ensembl IDs, when a human Ensembl ID maps to multiple mouse Ensembl IDs, the values will not get duplicated, but you will have multiple rows associated with that human Ensembl ID.
+Let's look at the `ENSG00000001617` example below.
+
+```{r}
+human_mouse_mapped_df %>%
+  dplyr::filter(human_ensembl == "ENSG00000001617")
+```
+
+## Collapse mouse genes mapping to multiple human genes
+
+Remember that if a mouse Ensembl gene ID maps to multiple human symbols, the values get duplicated.
+We can collapse the multi-mapped values into a list for each Ensembl ID as to not have duplicate values in our data frame.
+
+In the next chunk, we show how we can collapse all the human Ensembl IDs into one column where we store them all for each unique mouse Ensembl ID.
+
+```{r}
+collapsed_human_ensembl_df <- human_mouse_mapped_df %>%
+  # Group by mouse Ensembl IDs
+  dplyr::group_by(Gene) %>%
+  # Collapse the mapped values in `human_mouse_mapped_df` into one column named
+  # `all_human_ensembls` -- note that we will lose the `support` column in this summarizing step
+  dplyr::summarize(all_human_ensembls = paste(human_ensembl, collapse = ";"))
+
+head(collapsed_human_ensembl_df)
+```
+
+### Write results to file
+
+Now let's write our results to file.
+
+```{r}
+readr::write_tsv(
+  collapsed_human_ensembl_df,
+  file.path(
+    results_dir,
+    # Replace with a relevant output file name
+    "SRP070849_mouse_ensembl_to_collapsed_human_gene_symbol.tsv"
+  )
+)
+```
+
+## Collapse human genes mapping to multiple mouse genes
+
+Since multiple mouse Ensembl gene IDs map to the same human Ensembl gene ID, we may want to identify which one of these mappings represents the "true" ortholog, i.e. which mouse gene is most similar to the human gene we are interested in.
+*This is not at all straightforward!* (see, for just one example, https://doi.org/10.1093/bioinformatics/btaa468) [@Stamboulian2020].
```

Comment:
Probably should stick to our general format for links. 
```suggestion
*This is not at all straightforward!* (see, for just one example [this paper](https://doi.org/10.1093/bioinformatics/btaa468)) [@Stamboulian2020].
```

Reply:
*This is not at all straightforward!* (See [this paper](https://doi.org/10.1093/bioinformatics/btaa468) for just one example [@Stamboulian2020].)